devtools/create_installer_packages.rst

```diff
@@ -3,199 +3,195 @@
 
-A Conan package for a tool follows always a similar structure. This is a recipe for packaging the ``nasm`` tool for building assembler:
+
+.. note::
+
+    Conan v1.24 introduced a new feature to declare a full profile for the ``build`` and the ``host`` machine,
+    it is the preferred way to deal with this scenario. Older versions should rely on the deprecated
+    settings ``os_build`` and ``arch_build``. There is a small section below about those settings, for a full
+    explanation read the docs matching your Conan client.
+
+
+A Conan package for a tool is like any other package with an executable. Here it is a recipe for packaging
+the ``nasm`` tool for building assembler:
 
 .. code-block:: python
 
    import os
-   from conans import ConanFile
-   from conans.client import tools
+   from conans import ConanFile, tools
+   from conans.errors import ConanInvalidConfiguration
 
 
    class NasmConan(ConanFile):
        name = "nasm"
-       version = "2.13.01"
+       version = "2.13.02"
        license = "BSD-2-Clause"
        url = "https://github.com/conan-community/conan-nasm-installer"
-       settings = "os_build", "arch_build"
-       build_policy = "missing"
+       settings = "os", "arch"
        description="Nasm for windows. Useful as a build_require."
 
        def configure(self):
-           if self.settings.os_build != "Windows":
-               raise Exception("Only windows supported for nasm")
+           if self.settings.os != "Windows":
+               raise ConanInvalidConfiguration("Only windows supported for nasm")
 
        @property
        def nasm_folder_name(self):
            return "nasm-%s" % self.version
 
        def build(self):
-           suffix = "win32" if self.settings.arch_build == "x86" else "win64"
+           suffix = "win32" if self.settings.arch == "x86" else "win64"
            nasm_zip_name = "%s-%s.zip" % (self.nasm_folder_name, suffix)
            tools.download("http://www.nasm.us/pub/nasm/releasebuilds/"
                           "%s/%s/%s" % (self.version, suffix, nasm_zip_name), nasm_zip_name)
-           self.output.warn("Downloading nasm: "
+           self.output.info("Downloading nasm: "
                             "http://www.nasm.us/pub/nasm/releasebuilds"
                             "/%s/%s/%s" % (self.version, suffix, nasm_zip_name))
            tools.unzip(nasm_zip_name)
            os.unlink(nasm_zip_name)
 
        def package(self):
-           self.copy("*", dst="", keep_path=True)
+           self.copy("*", src=self.nasm_folder_name, dst="bin", keep_path=True)
            self.copy("license*", dst="", src=self.nasm_folder_name, keep_path=False, ignore_case=True)
 
        def package_info(self):
-           self.output.info("Using %s version" % self.nasm_folder_name)
-           self.env_info.path.append(os.path.join(self.package_folder, self.nasm_folder_name))
+           self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
```

Comment:
@jgsogo it should be `cpp_info.bindir`, no?

Reply:
I mean Iirc all helpers should use `run_environment` or virtualrunenv, so having `bindirs` should be enough. Did this change?

Reply:
Depending on where you use this recipe. If you use it with the new model with the two profiles, you just need to set `cpp_info.bindirs` to the `bin` directory (it is added by default). But if you want to use this same recipe using only one profile in the command line, you need to add the  `bin` folder to the PATH.

My intention here was to write an example that works with the two scenarios, even if this is talking about the two profiles... do you think it is misleading? I can imagine people copy/pasting an example and IMO it is better if it works for the two scenarios (if it requires almost no extra code)

Reply:
I mean Conan handled well ld_library_path on Linux, path on windows (for dynamic libraries) as well as macos wieredness with environment jamming in some cases.

And it felt that use of environment variables for path/dll path was discouraged or obsolete, things just worked (with `run_environment` though).

Reply:
🤔 You are right, `RunEnvironment` is already adding bin directories to `PATH`, so it works with one profile (this is what `run_envrionment=True` does) and with two profiles (logic that propagates environment from _build_ to _host_ does it already).

So this line is useless.

Reply:
Needed for the `virtualenv` generator (using one profile)

Reply:
Yes, that's why I merged them into one (in our company): as soon as you want to run a tool you need runenv. And this is 109% times since we have compiler and python as Conan packages.

Reply:
I think`run_environment=True` could be a default for Conan 2.0. and virtualenv should behave as if running from recipe (thus all virtual* merged).
 
I don't see why it is a harmful default. And we get defaults wrong: explicit, const, noexcept? 😂

Reply:
Yes, Conan 2.0 will reconsiders the defaults for the `self.run(...)` and the environment that is populated before entering recipe functions like `build()`. We are also aware that all `virtualxxxenv` generators might be merged into one.

It's been many years of 1.x and our commitment to stability, we really are looking forward for the next major version and break some of the conventions.